internal/expr/typeinfo.go

```diff
@@ -117,6 +137,73 @@ func (mi *mapInfo) getAllMembers() ([]typeMember, error) {
 
 var _ typeInfo = &mapInfo{}
 
+type primitiveTypeInfo struct {
+	primitiveType reflect.Type
+}
+
+func (pti *primitiveTypeInfo) typ() reflect.Type {
+	return pti.primitiveType
+}
+
+func (pti *primitiveTypeInfo) typeMember(member string) (typeMember, error) {
+	if member != "" {
+		return nil, fmt.Errorf(`cannot specify member of primitive type %q`, pti.primitiveType.Name())
+	}
+	return primitiveType{primitiveType: pti.primitiveType}, nil
+}
+
+func (pti *primitiveTypeInfo) getAllMembers() ([]typeMember, error) {
+	return nil, fmt.Errorf(`primitive type used in invalid context`)
```

Comment:
Missing test for this error.

Reply:
This error is actually unreachable in the current code. This is another place where the abstraction does not quite fit the reality. I've changed it to an internal error for now, but when the refactoring is in we should have another look at this.